tools/test/h5repack/CMakeLists.txt

```diff
@@ -27,19 +27,19 @@ set (REPACK_COMMON_SOURCES
     ${HDF5_TOOLS_SRC_H5REPACK_SOURCE_DIR}/h5repack_verify.c
     ${HDF5_TOOLS_SRC_H5REPACK_SOURCE_DIR}/h5repack.c
 )
+# Add h5repack test executable
 add_executable (h5repacktest ${REPACK_COMMON_SOURCES} ${HDF5_TOOLS_TEST_H5REPACK_SOURCE_DIR}/h5repacktst.c)
 target_include_directories (h5repacktest
     PRIVATE "${HDF5_TOOLS_SRC_H5REPACK_SOURCE_DIR};${HDF5_TOOLS_ROOT_DIR}/lib;${HDF5_SRC_INCLUDE_DIRS};${HDF5_TEST_SRC_DIR};${HDF5_SRC_BINARY_DIR};$<$<BOOL:${HDF5_ENABLE_PARALLEL}>:${MPI_C_INCLUDE_DIRS}>"
 )
-if (BUILD_STATIC_LIBS)
+if (HDF5_BUILD_STATIC_TOOLS)
   TARGET_C_PROPERTIES (h5repacktest STATIC)
-  target_link_libraries (h5repacktest PRIVATE ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET})
+  target_link_libraries (h5repacktest PRIVATE ${HDF5_TOOLS_LIB_TARGET} ${HDF5_TEST_LIB_TARGET} h5repackgentest)
```

Comment:
Since the tool gentest object libraries are currently only built when `HDF5_BUILD_GENERATORS` is `ON`, the linking step below fails when it's `OFF`. I'm guessing the reason it passes in most of the workflows is because they use the cacheinit file which sets that `ON` by default. I would generally like to avoid the requirement of `HDF5_BUILD_GENERATORS=ON` in cases where we don't need it, but this PR seems like it will need that, even for regular non-VOL testing. Maybe we need to decide if the `HDF5_BUILD_GENERATORS` option is still useful or if we should just always build the generators? Or maybe we should always build the generator _libraries_ but only build the generator binaries when `HDF5_BUILD_GENERATORS=ON`.